cognitedata/reveal
# feat: Zoom to cursor when mouse-scrolling in 360 images

#### Type of change

![Feat](https://img.shields.io/badge/Type-Feat-green)  

#### Jira ticket :blue_book:


https://cognitedata.atlassian.net/browse/REV-659

## Description :pencil:


In the `StationaryCameraManager` we use FOV to zoom. I now calculate a ray to the mouse position before and after the FOV is changed, and use the arc between these vectors to rotate the camera so that the cursor stays in the same position. In order to keep the "up-vector" constant the rotation on the camera is set using `lookAt,` which will preserve the up direction.

Zoom is not reset between points, but this is something we want to do. I tried a simple implementation of this, but the alpha&position animation we do during transitions looks broken when the FOV is not animated.  I have made an additional Jira task for this.

https://cognitedata.atlassian.net/browse/REV-689

## How has this been tested? :mag:



I have tested this using Reveal Examples and the MAUI_A model. I have loaded 360 images, zoomed in/out and transitioned between points. To make sure input works as intended after 360 images, I have also tested input after exiting 360 mode.

Tested in Chrome & Firefox.

## Test instructions :information_source:



See above. [Details from Slack on how to load images](https://cognitedata.slack.com/archives/G0115JQ8JM8/p1669380585627129).

## Checklist :ballot_box_with_check:
